addons/website_product_comparison/static/src/js/website_product_comparison.js

```diff
@@ -0,0 +1,342 @@
+        if ($('.comparator-popover').length) {
+            $('#comparelist .o_product_panel_header').popover('show');
+        }
+    },
+    add_new_products:function(product_id){
+        var self = this;
+        self.$('.o_comparelist_warning').hide();
+        if (!_.contains(self.comparelist_product_ids, product_id)) {
+            self.comparelist_product_ids.push(product_id);
+            if(_.has(self.parent.product_data, product_id)){
+                self.update_content([product_id], false);
+            } else {
+                self.parent.load_products([product_id]).then(function(){
+                    self.update_content([product_id], false);
+                });
+            }
+        }
+    },
+    update_content:function(product_ids, reset) {
+        var self = this;
+        if (reset) {
+            self.$('.o_comparelist_products .o_product_row').remove();
+        }
+        if (product_ids.length) {
+            var category = self.check_product_category(self.parent.product_data[product_ids[0]]);
+            if (category) {
+                _.each(product_ids, function(res) {
+                    self.$('.o_product_panel_empty').hide();
+                    var $template = self.parent.product_data[res].render;
+                    self.$('.o_comparelist_products').append($template);
+                });
+                self.update_cookie();
+            }
+        }
+        this.refresh_panel();
+    },
+    check_product_category: function(data) {
+        var category_ids = $('#comparelist .o_product_row:first').data('category_ids');
+        if (category_ids){
+            var categories = JSON.parse("[" + category_ids + "]"); // TO DO : use parent
+            if (!_.intersection(categories, data.product.public_categ_ids).length) {
+                this.$('.o_comparelist_warning').show();
+                this.$('#comparelist_alert').text(data.product.name);
+                this.show_panel(true);
+                this.comparelist_product_ids = _.without(this.comparelist_product_ids, data.product.id);
+                return false;
+            }
+        }
+        $('.o_comparelist_warning').hide();
+        return true;
+    },
+    rm_from_comparelist: function(e){
+        this.comparelist_product_ids = _.without(this.comparelist_product_ids, $(e.currentTarget).data('product_product_id'));
+        $(e.currentTarget).parents('.o_product_row').remove();
+        this.$('.o_comparelist_warning, .o_comparelist_limit_warning').hide();
+        this.update_cookie();
+        // force refresh to reposition popover
+        this.update_content(this.comparelist_product_ids, true);
+    },
+    update_cookie: function(){
+        document.cookie = 'comparelist_product_ids=' + JSON.stringify(this.comparelist_product_ids) + '; path=/';
+        this.update_comparelist_view();
+    },
+    update_comparelist_view: function() {
+        this.$('.o_product_circle').text(this.comparelist_product_ids.length);
```

Comment:
This is the number of products in the cookie but not the number of displayed products
![selection_010](https://cloud.githubusercontent.com/assets/564822/23162412/2f819190-f82e-11e6-990c-c9de905823df.png)